frontend/javascripts/admin/dataset/dataset_add_zarr_view.tsx

```diff
@@ -0,0 +1,227 @@
+import { Form, Input, Button, Col, Row, Collapse } from "antd";
+import { connect } from "react-redux";
+import React, { useState } from "react";
+import type { APIUser } from "types/api_flow_types";
+import type { OxalisState } from "oxalis/store";
+import { exploreRemoteDataset, storeRemoteDataset } from "admin/admin_rest_api";
+import messages from "messages";
+import { jsonStringify } from "libs/utils";
+import { CardContainer } from "admin/dataset/dataset_components";
+import Password from "antd/lib/input/Password";
+import TextArea from "antd/lib/input/TextArea";
+import { AsyncButton } from "components/async_clickables";
+import Toast from "libs/toast";
+import DataLayer from "oxalis/model/data_layer";
+import _ from "lodash";
+import { Hint } from "oxalis/view/action-bar/download_modal_view";
+const { Panel } = Collapse;
+const FormItem = Form.Item;
+
+type OwnProps = {
+  onAdded: (arg0: string, arg1: string) => Promise<void>;
+};
+type StateProps = {
+  activeUser: APIUser | null | undefined;
+};
+type Props = OwnProps & StateProps;
+
+function DatasetAddZarrView(props: Props) {
+  const { activeUser, onAdded } = props;
+  const [datasourceConfig, setDatasourceConfig] = useState<string>();
+  const [exploreLog, setExploreLog] = useState<string>("");
+  const [datasourceUrl, setDatasourceUrl] = useState<string>("");
+  const [usernameOrAccessKey, setUsernameOrAccessKey] = useState<string>("");
+  const [passwordOrSecretKey, setPasswordOrSecretKey] = useState<string>("");
+  const [selectedProtocol, setSelectedProtocol] = useState<string>("https://");
+
+  function validateUrls(userInput: string) {
+    if (
+      (userInput.indexOf("https://") === 0 && userInput.indexOf("s3://") !== 0) ||
+      (userInput.indexOf("https://") !== 0 && userInput.indexOf("s3://") === 0)
+    ) {
+      setSelectedProtocol(userInput.indexOf("https://") === 0 ? "https://" : "s3://");
+    } else {
+      throw new Error("Dataset URL must employ either the https:// or s3:// protocol.");
+    }
+  }
+
+  async function handleExplore() {
+    if (datasourceUrl) {
+      const { dataSource, report } =
+        !usernameOrAccessKey || !passwordOrSecretKey
+          ? await exploreRemoteDataset([datasourceUrl])
+          : await exploreRemoteDataset([datasourceUrl], {
+              username: usernameOrAccessKey,
+              pass: passwordOrSecretKey,
+            });
+      setExploreLog(report.toString());
+      if (dataSource) {
+        if (datasourceConfig) {
+          let loadedDatasource;
+          try {
+            loadedDatasource = JSON.parse(datasourceConfig);
+          } catch (e) {
+            Toast.error("The loaded datasource config contains invalid JSON.");
+            return;
+          }
+          if (_.isEqual(loadedDatasource.scale, dataSource.scale)) {
+            Toast.warning(
+              `${messages["dataset.add_zarr_different_scale_warning"]}\n${dataSource.scale.join(
+                "\n",
```

Comment:
```suggestion
                ", ",
```

Reply:
I would like to leave this as it it more readable to me to have the scale like this:
123
123
123
Also since the warning will disappear again quite fast.

Reply:
You could pass `{ timeout: 10000 }` as a second parameter to `Toast.warning` to make the toast stay longer. Having the scale in three lines is quite unusual in wk. We never really do this :see_no_evil: Also, it would be helpful to have the conflicting second scale in the message, too. How about this:
```
The actual error message comes here:        // <-- note the line break
24, 24, 24 != 11,11,8
```

With the line break, the scales would appear in a new line and should be easy to spot. What do you think?

Reply:
The reason I separated the values with \n was that some scales are rather long floating point values which will wrap around and become difficult to read. But I'm also fine with this approach 👍 
E. g.:
"scale": [
    11.239999771118164,
    11.239999771118164,
    28
  ]

Reply:
ah, interesting. then, I'd recommend the `formatScale` function from `format_utils.ts`. that function will take care of rounding these numbers (and it will also turn the array into a string).